pkg/operator/ceph/object/objectstore_test.go

```diff
@@ -815,3 +816,98 @@ func TestGetRealmKeyArgs(t *testing.T) {
 		assert.Equal(t, "", secret)
 	})
 }
+
+func TestUpdateZoneEndpointList(t *testing.T) {
+	type args struct {
+		zoneEndpointList  []string
+		undesiredEndpoint string
+		desiredEndpoint   string
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{"all the fields are empty",
+			args{zoneEndpointList: []string{}, undesiredEndpoint: "", desiredEndpoint: ""},
+			[]string{},
+		},
+		{"emptylist - old value exists",
+			args{zoneEndpointList: []string{}, undesiredEndpoint: "http://rgw-endpoint", desiredEndpoint: ""},
+			[]string{},
+		},
+		{"adding new endpoint to emptylist",
+			args{zoneEndpointList: []string{}, undesiredEndpoint: "", desiredEndpoint: "http://rgw-endpoint"},
+			[]string{"http://rgw-endpoint"},
+		},
+		{"emptylist - old and new are same",
+			args{zoneEndpointList: []string{}, undesiredEndpoint: "http://rgw-endpoint", desiredEndpoint: "http://rgw-endpoint"},
+			[]string{"http://rgw-endpoint"},
+		},
+		{"emptylist - old and new value different values",
+			args{zoneEndpointList: []string{}, undesiredEndpoint: "http://old-endpoint", desiredEndpoint: "http://new-endpoint"},
+			[]string{"http://new-endpoint"},
+		},
+		{"adding new endpoint to existing list",
+			args{zoneEndpointList: []string{"http://rgw-endpoint-1"}, undesiredEndpoint: "", desiredEndpoint: "http://new-endpoint"},
+			[]string{"http://rgw-endpoint-1", "http://new-endpoint"},
+		},
+		{"removing old endpoint without updating new value",
+			args{zoneEndpointList: []string{"http://rgw-endpoint-1"}, undesiredEndpoint: "http://rgw-endpoint-1", desiredEndpoint: ""},
+			[]string{},
+		},
+		{"removing endpoint which is not present in the list without updating new value",
+			args{zoneEndpointList: []string{"http://rgw-endpoint-1"}, undesiredEndpoint: "http://rgw-endpoint-2", desiredEndpoint: ""},
+			[]string{"http://rgw-endpoint-1"},
+		},
+		{"no updation or removal from list",
+			args{zoneEndpointList: []string{"http://rgw-endpoint-1"}, undesiredEndpoint: "", desiredEndpoint: ""},
+			[]string{"http://rgw-endpoint-1"},
+		},
+		{"updating same endpoint to list",
+			args{zoneEndpointList: []string{"http://rgw-endpoint-1"}, undesiredEndpoint: "http://rgw-endpoint-1", desiredEndpoint: "http://rgw-endpoint-1"},
+			[]string{"http://rgw-endpoint-1"},
+		},
```

Comment:
What about the longer zoneEndpointList case where undesired and desired are the same?

IMO, this would be desired behavior for this case:
```
updateZoneEndpointList([]string{"e1", "e2", "e3"}, "e2", "e2")
  -> []string{"e1", "e2", "e3"}
```

This seems like a valuable test to me, because I think it probably makes sense that we don't want to change the order of the endpoints if we don't have to. I'm not sure if Ceph cares if the endpoints get re-ordered, but some systems might allow the no-update case to be a non-disruptive one.